internal/pgdump/pgdump.go

```diff
@@ -0,0 +1,64 @@
+package pgdump
+
+import (
+	"fmt"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+// Targets represents configuration for each of Sourcegraph's databases.
+type Targets struct {
+	Primary      Target `yaml:"primary"`
+	CodeIntel    Target `yaml:"codeintel"`
+	CodeInsights Target `yaml:"codeinsights"`
+}
+
+// Target represents a database for pg_dump to export.
+type Target struct {
+	// e.g. container, deployment, statefulset, etc.
+	Target string `yaml:"entity"`
```

Comment:
```suggestion
	// Target is the name of the database deployment
	// When built-in database is used,
	// docker: the name of the database container, e.g. pgsql, codeintel-db, codeinsights-db
	// k8s: the name of the deployment or statefulset, e.g. deploy/pgsql, sts/pgsql, 
	Target string `yaml:"entity"`
```

I think we could also interpret `target` as hostname of the database host. for example, a customer could be using an external database

Reply:
also, should we rename the key name in yaml to `target` instead of `entity`? I feel like the word `entity` is a bit confusing

Reply:
Ah yes the yaml should definitely be `target`!